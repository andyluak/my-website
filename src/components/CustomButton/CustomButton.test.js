import React from 'react';
import { shallow, render } from 'enzyme';
import CustomButton from './CustomButton';

describe( 'Custom button component tests', () =>{
	it( 'excepct to render Custom Button component', () => {
		expect(shallow(<CustomButton />)).toMatchSnapshot();
	})

	it( 'excepct to render Custom Button component with children', () => {
		expect(render(<CustomButton>Hello</CustomButton>)).toMatchSnapshot();
	})

	it( 'excepct to render Custom Button component with children and props', () => {
		expect(render(<CustomButton disabled>Hello</CustomButton>)).toMatchSnapshot();
	})

	it( 'excepct to render Custom Button component with children and type submit', () => {
		expect(render(<CustomButton type="submit">Hello</CustomButton>)).toMatchSnapshot();
	})
})