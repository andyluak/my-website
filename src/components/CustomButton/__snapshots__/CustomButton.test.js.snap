// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom button component tests excepct to render Custom Button component 1`] = `
<button
  className="custom-button"
/>
`;

exports[`Custom button component tests excepct to render Custom Button component with children 1`] = `
<button
  class="custom-button"
>
  Hello
</button>
`;

exports[`Custom button component tests excepct to render Custom Button component with children and props 1`] = `
<button
  class="custom-button"
  disabled=""
>
  Hello
</button>
`;

exports[`Custom button component tests excepct to render Custom Button component with children and type submit 1`] = `
<button
  class="custom-button"
  type="submit"
>
  Hello
</button>
`;
